Content.Client/Clothing/ClothingSystem.cs

```diff
@@ -0,0 +1,170 @@
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.ResourceManagement;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.IoC;
+
+namespace Content.Client.Clothing;
+
+public class ClothingSystem : EntitySystem
+{
+    /// <summary>
+    /// This is a shitty hotfix written by me (Paul) to save me from renaming all files.
+    /// For some context, im currently refactoring inventory. Part of that is slots not being indexed by a massive enum anymore, but by strings.
+    /// Problem here: Every rsi-state is using the old enum-names in their state. I already used the new inventoryslots ALOT. tldr: its this or another week of renaming files.
+    /// </summary>
+    private static readonly Dictionary<string, string> TemporarySlotMap = new()
+    {
+        {"head", "HELMET"},
+        {"eyes", "EYES"},
+        {"ears", "EARS"},
+        {"mask", "MASK"},
+        {"outerClothing", "OUTERCLOTHING"},
+        {"jumpsuit", "INNERCLOTHING"},
+        {"neck", "NECK"},
+        {"back", "BACKPACK"},
+        {"belt", "BELT"},
+        {"gloves", "HAND"},
+        {"shoes", "FEET"},
+        {"id", "IDCARD"},
+        {"pocket1", "POCKET1"},
+        {"pocket2", "POCKET2"},
+    };
+
+    [Dependency] private IResourceCache _cache = default!;
+    [Dependency] private InventorySystem _inventorySystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ClothingComponent, GotEquippedEvent>(OnDidEquip);
+        SubscribeLocalEvent<ClothingComponent, GotUnequippedEvent>(OnDidUnequip);
+    }
+
+    public void InitClothing(EntityUid uid, ClientInventoryComponent? component = null, SpriteComponent? sprite = null)
+    {
+        if (!_inventorySystem.TryGetSlots(uid, out var slots, component) || !Resolve(uid, ref sprite, ref component)) return;
+
+        foreach (var slot in slots)
+        {
+            sprite.LayerMapReserveBlank(slot.Name);
+
+            if (!_inventorySystem.TryGetSlotContainer(uid, slot.Name, out var containerSlot, out _, component) ||
+                !containerSlot.ContainedEntity.HasValue) continue;
+
+            RenderEquipment(uid, containerSlot.ContainedEntity.Value, slot.Name, component, sprite);
+        }
+    }
+
+    private void OnDidUnequip(EntityUid uid, ClothingComponent component, GotUnequippedEvent args)
+    {
+        SpriteComponent? sprite = null;
+        if (!Resolve(args.Equipee, ref sprite))
+        {
+            return;
+        }
+
+        sprite.LayerSetVisible(args.Slot, false);
+    }
+
+    private void OnDidEquip(EntityUid uid, ClothingComponent component, GotEquippedEvent args)
+    {
+        if (!TryComp<SpriteComponent>(args.Equipee, out var sprite) || !TryComp<ClientInventoryComponent>(args.Equipee, out var invComp))
+        {
+            return;
+        }
+
+        if (TryComp(args.Equipment, out ClothingComponent? clothing))
+        {
```

Comment:
If I understand correctly, here `uid == args.Equipment` and `component == clothing`, so the TryGet is unnecessary?

Also, the having a function called `OnDidEquip` handling a `GotEquippedEvent` is a bit confusing. Maybe rename it to OnGotEquipped?

Reply:
this is a blunder on my part ye